src/test/java/io/confluent/connect/jdbc/source/JdbcSourceTaskUpdateTest.java

```diff
@@ -342,10 +423,38 @@ public void testAutoincrementRestoreOffset() throws Exception {
   }
 
   @Test
-  public void testTimestampRestoreOffset() throws Exception {
+  public void testTimestampRestoreNoVersionOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), null);
+    testTimestampRestoreOffset(Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap()));
+  }
+
+  @Test
+  public void testTimestampRestoreVersionOneOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), null);
+    testTimestampRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap())
+    );
+  }
+
+  @Test
+  public void testTimestampRestore() throws Exception {
```

Comment:
This is testing when we have offsets with both protocols. Is that worth having the method include that information? E.g., `testTimestampRestoreOffsetsWithMultipleProtocol()`